### DISCUSSION TITLE
Is it possible to merge the output of tee?

### DISCUSSION THREAD

**Discussion by huntc** [nushell/nushell#13619] [Q&A]
Is it possible to merge the output of the new pipeline created by `tee` back into the original pipeline?

For example, given this contrived example:

```
1..100 | tee { each { |v| print ($v * 2) } } | math sum | wrap sum
```

...could I sum the pipeline output of the `tee` into the outer pipeline, effectively merging them back in?

**Comment by ghost** [comment]
you could try your luck with mutables and write a lot of handling, but i usually use a non-nushell-native approach: `fifo`

```nu
# create fifo
let fifo_dir = (mktemp -d)  # mkfifo needs to create the file itself -> tmpdir and not tmpfile
let fifo = $"($tmpdir)/fifo"
^mkfifo $fifo

# write initial values
1..100 | collect | str join "\n" | save -a $fifo

# the actual loop
^cat $fifo  # `open` might work, but with streams `cat` is a safer bet
| lines
| each {|i|
  $i | save -a $fifo  # write back
}

# clean up
rm -r $fifo_dir
```

this example just creates a infinite useless loop, but shows how to do it (i hope).

if you want structured data you will have to add something like `to nuon` and `from nuon`

this is windows-compatible, but i think windows has a `fifo` equivalent somewhere.

**Comment by huntc** [comment]
Thanks. I think that's a lot of ceremony though. I actually have control over the downstream tool that I'm presently invoking `tee` on, so I think I'll just change that to filter out the field it wants from a stream of JSON.

That aside, do we think combining pipelines might be a feature worth integrating with nushell?
